### PR TITLE
Cleanup thanos ruler

### PR DESCRIPTION
**Pull request by simonpasquier**
~~This needs #696 first.~~

### CODE REVIEW

**Comment by s-urbaniak** [comment]
looking good so far :+1: needs a rebase

File: assets/thanos-ruler/service-monitor.yaml [hunk 1]
```diff
@@ -2,17 +2,18 @@ apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
   labels:
-    k8s-app: alertmanager
+    k8s-app: thanos
```
Comment (lilic) [inline assets/thanos-ruler/service-monitor.yaml]: should this not be thanos-ruler instead?

File: jsonnet/thanos-ruler.jsonnet [hunk 2]
```diff
@@ -191,19 +158,18 @@ local authorizationRole =
           name: 'thanos-ruler',
           namespace: thanosRulerConfig.namespace,
           labels: {
-            'k8s-app': 'alertmanager',
+            'k8s-app': 'thanos',
```
Comment (lilic) [inline jsonnet/thanos-ruler.jsonnet]: Same question here.

Comment (simonpasquier) [reply jsonnet/thanos-ruler.jsonnet]: :+1:

**Comment by lilic** [comment]
>         user_workload_monitoring_test.go:92: timed out waiting for the condition: statefulsets.apps "prometheus-user-workload" not found

Seems like a valid failure. 🤔

**Comment by simonpasquier** [comment]
@lilic yes the operator was using the now removed asset.

**Comment by lilic** [comment]
/retest

Unrelated failure.

**Pull request merged**